### DISCUSSION TITLE
图床

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4306] [插件说明]
描述
---

可以将阅读模式的图片上传到你的图床中，并替换为图床的图片链接，导入后支持触发自动化方案实现导入到简悦支持的任意生产力工具的诉求，目前有以下几种导入方式：

1️⃣ PicGo 图床 → 借助 PicGo App 实现支持大多数常见的图床方案。

2️⃣ PicList 图床 → 借助 PicList App 实现支持大多数常见的图床方案。

3️⃣ Base64 本地化图片 → 将图片转换为 Base64 字符串。

4️⃣ Cloudinary 图床 → 直接将图片上传到 Cloudinary 图床。

地址
---

[https://simpread.ksria.cn/plugins/details/VdMfVaXlni](https://simpread.ksria.cn/plugins/details/VdMfVaXlni)

快捷键
---

`f f`

视频
---

https://github.com/Kenshin/simpread/assets/81074/17d0a27a-af5f-4cc3-864d-8c6185ab3102

选项
---

随便打开一个页面 e.g. https://sspai.com/post/69972 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1419" alt="image" src="https://github.com/user-attachments/assets/ec50c194-e20f-458b-be9c-55db3fea75a3">

功能
---

1. 支持图床方案：PicGo 图床、PicList 图床、[Cloudinary 图床](https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-7172818) 与 **Base64 图片本地化** 三种方式。

2. 图片上传完毕后自动将阅读模式的图片链接转换为图床链接。

3. 图片上传完毕后自动加入稍后读

     <img width="540" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/e813e978-9af7-4a15-a58b-8af86ca5cdee">

4. 图片上传完毕后支持导出方案（包括：简悦扩展端原生支持与简悦插件）

    <img width="703" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/72822fe8-aecc-4840-819f-347dbecbe388">

5. 仅将图片转换为 Picgo 图床 /  PicList 图床 / Cloudinary 图床 / Base64 格式

    <img width="276" alt="image" src="https://github.com/user-attachments/assets/159a844b-31ff-4fa8-a1a9-86033f445bb8">

PicGo 图床
---

[PicGo](https://picgo.github.io/PicGo-Doc/) 是一个包含全平台的图床上传工具，并且在 `2.2.0` 版增加了 `PicGo-Server` 功能，所以简悦集成了此功能，进而非常快速的解决了图床问题。

需要在 PicGo 设置你需要的图床，细节可以看 [PicGo 的设置](https://picgo.github.io/PicGo-Doc/zh/guide/config.html#图床区)，并将此图床设置为默认值。

PicList 图床
---

[PicList](https://piclist.cn/) PicList 是在 PicGo 的基础上经过深度的二次开发的第三方图床工具，相比 PicoGo 解决了某些网站图片无法上传的问题，如： 少数派、微信公众号等。

需要在 PicList 设置你需要的图床，细节可以看 [PicList 的设置](https://picgo.github.io/PicGo-Doc/zh/guide/config.html#图床设置区)，然后按下图所示复制 API

<img width="1003" alt="image" src="https://github.com/user-attachments/assets/8d57d498-d828-400a-b86e-a08fc5d67089">

并粘贴到此处

<img width="660" alt="image" src="https://github.com/user-attachments/assets/c8cb8c5e-2e40-41a4-819e-b6391edcfa4c">

然后验证设置是否正确并保存，即可。

Cloudinary 图床
---

与 [导入到 Notion 插件](https://github.com/Kenshin/simpread/discussions/4954) 内置一样的方案，详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-7172818)。

具体例子请看 [利用图床插件内置的 Cloudinary 图床解决个别网站无法转换图片的问题（进而解决无法使用 assets + md 方案或 Textbundle 方案的问题）](https://github.com/Kenshin/simpread/discussions/6598)

Base64 图片本地化
---

将阅读模式的图片转换为 `base64` 格式（这是一种图片字符串化的方案），无需借助任意图床，是一种比较灵活的图片本地化方案。

具体例子请看 [使用简悦提供的【离线下载辅助增强插件】与【图床插件】解决在导出时图片无法显示的问题 ](https://github.com/Kenshin/simpread/discussions/6597)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/VdMfVaXlni) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

关联
---

- [使用简悦提供的【离线下载辅助增强插件】与【图床插件】解决在导出时图片无法显示的问题](https://github.com/Kenshin/simpread/discussions/6597)

更新日志
---

### 2.2.1

- `Add` 阅读模式右下角 → 插件 → 插件触发器 → **将图片导入到图床**

- `Add` 版本检查（选项卡左下角版本号，点击可验证是否有新版）

- `Update` 选项卡升级到 2.0

### 2.2.0

- `Add` PicList 图床 https://github.com/Kenshin/simpread/issues/6733

### 2.1.0

- `Add` [内置 Cloudinary 账户](https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-9995715)

### 2.0.0

1. `Add` 在 PicGo 的基础上，内置了 [Cloudinary 图床](https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-7172818) 与 **Base64 图片本地化** 两种方案。

2. `Add` 图片上传完毕后自动加入稍后读

3. `Add` 仅将图片转换为 Picgo 图床 /  Cloudinary 图床 / Base64 格式

### 1.0.0

1. 将当前页面的全部图片转换为在 PicGo 设定好的图床。

2. 将当前阅读模式的图片链接转换为图床链接。

3. 当导入后会自动触发预设服务。

**Comment by andyboyce** [comment]
这个功能可以稍后阅读或者标记谋篇文章的时候，就能把文章了里面的图片全部上传到自己的图床，然后同步到obsidian等的html里面的文章里面的图片也是自己图床的图片吗？（这样一般就不用去搞离线html了）

**Reply by Kenshin** [reply]
> 标记谋篇文章的时候，就能把文章了里面的图片全部上传到自己的图床

标记图片的时候，确实没有这个功能，但思路不错，可以考虑。

***

换个角度说，可以使用这个插件实现，当转换为自己的图床图片后，自动加入稍后读。

**Comment by Kenshin** [comment]
# Cloudinary 图床

[Cloudinary](https://cloudinary.com/) 是老牌的图床服务，**免费版每月 25 Credit，每个 Credit 可支持 1GB 存储或 1GB 带宽**。

相当于：**每月 13GB 存储**以及 **12GB 带宽**的图床，且**国内访问速度不错**，免费份额足够个人使用。

## 注册

与 **导入到 Notion 插件** 内置一样的方案，如何注册和使用 [请看这里](https://github.com/Kenshin/simpread/discussions/4954#discussioncomment-7091995) 。

## 配置

注册并登录后，打开直接登录进入 [Dashboard 页面](https://console.cloudinary.com/console)  后得到 `name` `key` `secret` 这三个值

<img width="1251" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/1052685d-95e3-4f75-85cf-9d463ef3b7ac">

然后设置一次复制到下图所示位置

<img width="542" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/233c0d86-22fe-4e52-a7c1-3ef330d35dfb">

保存后，可以验证是否成功

<img width="721" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/ce6698bb-4ebe-4864-bbf0-27e59549718d">

验证成功时的提示

![2023-09-26_12-29-53 (1)](https://github.com/Kenshin/simpread/assets/81074/98c6f450-debd-4c62-8f03-47dad2188845)

## 使用

需要开启 Cloudinary 图床

<img width="510" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/b7fa84db-b0a1-435e-aeed-962b1370ed2c">

## 其它功能

### 导入到 Cloudinary 的方案

https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-9995714

### 内置 Cloudinary 账户

https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-9995715

**Reply by Kenshin** [reply]
# 导入到 Cloudinary 的方案

旧版 1.2.0（以上）会通过简悦的自己的服务器，优点是不需要耗费用户的设备性能；缺点是当为内网图片时无法使用；
新版 1.2.0（含），不经过简悦自己的图片服务器，解决了内网图片或某些有防盗链的图片无法上传的问题。

<img width="521" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/b39f35f7-763b-4e71-b093-7ccde0114990">

## 权限

此方式需要简悦的 [高级账户](https://simpread.pro/price) 权限。

**Reply by Kenshin** [reply]
# 内置 Cloudinary 账户

2.1.0 版 **内置了 Cloudinary 账户，仅做测试时使用，不建议在正式环境中使用**，此账户内的图片会被公开，当流量封顶时会停止服务。

<img width="537" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/9e92099c-ed95-461c-82f3-2043c4e38979">

**Reply by gladego** [reply]
K 哥 @Kenshin ，能不能帮我看看是哪里出了问题，我注册 Cloudinary 并填写了配置 3 项，点击验证 Cloudinary 图床，右上角显示上传成功。如图一所示。

但我点击导出后，右上角一直显示导入失败，你觉得可能是哪里出了问题？如图二所示。

我已经尝试的解决方法：
- 更换 Cloudinary API
- 使用简悦官方内置 Cloudinary 账号导出，依然失败，如图三所示

<img width="1280" height="594" alt="PixPin_2026-09-04_12-23-00" src="https://github.com/user-attachments/assets/7ede42c1-814f-44f2-9c20-33c8f9cf0b8a" />
<img width="1280" height="594" alt="PixPin_2026-09-04_12-31-31" src="https://github.com/user-attachments/assets/673ae286-66f1-4289-954c-6690cd42d9bc" />
<img width="1280" height="594" alt="PixPin_2026-09-04_12-34-44" src="https://github.com/user-attachments/assets/5f2a9dae-7289-4458-a5d6-a975087a7024" />

**Comment by Kenshin** [comment]
# 图片转换完毕后自动加入到稍后读

<img width="537" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/f6de70f0-4dc5-4963-9831-47b675a22932">

## 演示

使用快捷键 <kbd>f</kbd> <kbd>f</kbd>，将图片转换为 Cloudinary 图床，然后自动加入稍后读的演示。


注意：因为使用了图床方案，所以不需要使用 <kbd>d</kbd> <kbd>d</kbd> 方案。

https://github.com/Kenshin/simpread/assets/81074/17d0a27a-af5f-4cc3-864d-8c6185ab3102

不支如此，此方式还包含了：加入到稍后读自动导入到 Obsidian 的方式。（留意导入到 Obsidian 的图片链接是你的 Cloudinary 图床

<img width="977" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/c5a0903e-20fd-4645-9ac6-35ce81c6e63e">

通过此步骤就可以将图片彻底永久化（在 Cloudinary 图床），不仅（自动化加入稍后读），还包含更多导出方案 https://github.com/Kenshin/simpread/discussions/4306#discussioncomment-7172968

**Comment by Kenshin** [comment]
# 支持任意导出服务

不仅（自动化加入稍后读），还包含更多导出方案（更灵活）

<img width="719" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/1d22206d-81da-4bc0-83f6-be5cf0826229">

## 注意

如果想要使用此方式，需要 **【关闭】自动加入稍后读** 的功能。

<img width="538" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/6566cabd-3e43-419f-9c00-e40c02c82fd3">

## 与【自动加入稍后读】的区别

如果只是单纯的希望将某个阅读模式的图片转换为图床，然后下载为 HTML 或 Markdown 或简悦支持的导出服务，不需要加入稍后读的话，则可以使用这个方式。

此方式比 **自动加入稍后读** 更加灵活。
